src/GitHub.App/Services/RepositoryCloneService.cs

```diff
@@ -103,6 +106,24 @@ public async Task<ViewerRepositoriesModel> ReadViewerRepositories(HostAddress ad
             return result;
         }
 
+        /// <inheritdoc/>
+        public async Task CloneOrOpenRepository(
+            UriString url,
+            string repositoryPath,
+            object progress = null)
+        {
+            Guard.ArgumentNotEmptyString(url, nameof(url));
+            Guard.ArgumentNotEmptyString(repositoryPath, nameof(repositoryPath));
+
```

Comment:
Should we be checking whether the repository at this path has the same clone URL? I'm thinking of the case where e.g. the user wants to clone a fork and they already have the parent repository at this location.

Reply:
Yes! In fact there are a few scenarios we should be checking for.

1. That a repository exists at the target path
2. That the repository has an `origin` remote
3. That the repository's remote matches the target URL

I'll let you know when these checks land.